pkg/kubectl/cmd/set/set_serviceaccount.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package set
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/util/i18n"
+)
+
+var (
+	serviceaccountResources = `
+	replicationcontroller (rc), deployment (deploy), daemonset (ds), job, replicaset (rs), statefulset`
+
+	serviceaccountLong = templates.LongDesc(`
+	Update ServiceAccount of pod template resources.
+
+	Possible resources (case insensitive) can be: 
+	` + serviceaccountResources)
+
+	serviceaccountExample = templates.Examples(`
+	# Set Deployment nginx-deployment's ServiceAccount to serviceaccount1
+	$kubectl set service-account deployment nginx-deployment serviceaccount1
+
+	# Short form of the above command
+	$kubectl set sa deploy nginx-deployment serviceaccount1
+
+	# Print result in yaml format of updated nginx deployment from local file, without hitting apiserver
```

Comment:
How about this `Print the result (in yaml format) of updated nginx deployment with serviceaccount from local file..` ?   just add brackets is also ok, I'm not persist, your choice